packages/cli/src/commands/run/filterPipelineDefinitions.ts

```diff
@@ -0,0 +1,18 @@
+import type { Target } from "@lage-run/target-graph";
+import type { PipelineDefinition } from "../../types/PipelineDefinition";
+
+export function filterPipelineDefinitions(targets: IterableIterator<Target>, pipeline: PipelineDefinition) {
+  const tasksSet = new Set<string>();
+
+  for (const target of targets) {
+    tasksSet.add(target.task);
+  }
+  const filteredPipeline: PipelineDefinition = {};
+  for (const [id, definition] of Object.entries(pipeline ?? {})) {
```

Comment:
pipeline ?? {}

I'm missing something here - what happens when the pipeline is null?

Reply:
pipeline being null would just set some defaults for you. default mode for tasks is "task: [^task]" I think